### ISSUE TITLE
About Tabbar switching prints nearby Tabbar issues

### ISSUE THREAD

**Issue by ahyangnb** [flutter/flutter#51154]
Why does the Tabbar switch print the Tab pages near the TabBarView that I want to switch, for example, I would print from 0 to 5: 4, 5, 4, and then print from 5 to 1: 2, 1, 2,

Asked the group, they said that this is the same as Android, but asked if this should still happen.

This is my code
```dart
import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: '',
      home: new MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> with TickerProviderStateMixin {
  TabController tabC;

  List tabs = [0, 1, 2, 3, 4, 5];

  @override
  void initState() {
    super.initState();
    tabC = new TabController(length: tabs.length, vsync: this);
  }

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(
        title: new Text('test Tabbar'),
        bottom: new TabBar(
          controller: tabC,
          tabs: tabs.map((tab) => new Tab(text: '$tab')).toList(),
        ),
      ),
      body: new TabBarView(
        controller: tabC,
        children: [
          new Page0(),
          new Page1(),
          new Page2(),
          new Page3(),
          new Page4(),
          new Page5(),
        ],
      ),
    );
  }
}

class Page0 extends StatefulWidget {
  @override
  _Page0State createState() => _Page0State();
}

class _Page0State extends State<Page0> {
  @override
  void initState() {
    super.initState();
    print('0');
  }

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

class Page1 extends StatefulWidget {
  @override
  _Page1State createState() => _Page1State();
}

class _Page1State extends State<Page1> {
  @override
  void initState() {
    super.initState();
    print('1');
  }

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

class Page2 extends StatefulWidget {
  @override
  _Page2State createState() => _Page2State();
}

class _Page2State extends State<Page2> {
  @override
  void initState() {
    super.initState();
    print('2');
  }

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

class Page3 extends StatefulWidget {
  @override
  _Page3State createState() => _Page3State();
}

class _Page3State extends State<Page3> {
  @override
  void initState() {
    super.initState();
    print('3');
  }

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

class Page4 extends StatefulWidget {
  @override
  _Page4State createState() => _Page4State();
}

class _Page4State extends State<Page4> {
  @override
  void initState() {
    super.initState();
    print('4');
  }

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

class Page5 extends StatefulWidget {
  @override
  _Page5State createState() => _Page5State();
}

class _Page5State extends State<Page5> {
  @override
  void initState() {
    super.initState();
    print('5');
  }

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}
```

My environment:
```
[✓] Flutter (Channel unknown, v1.12.13, on Mac OS X 10.14.5 18F2059, locale
    zh-Hans-CN)
    • Flutter version 1.12.13 at /Users/q1/flutter
    • Framework revision cf37c2cd07 (3 months ago), 2019-11-25 12:04:30 -0800
    • Engine revision b6b54fd606
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/q1/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/q1/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (3 available)
    • iPhone 8   • 5DEF137E-B0E4-4DE0-BEA3-31341A5B1B08 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
    • Chrome     • chrome                               • web-javascript •
      Google Chrome 80.0.3987.116
    • Web Server • web-server                           • web-javascript •
      Flutter Tools

• No issues found!
```

**Comment by jonahwilliams** [comment]
Questions like these are better suited for Q&A sites like [StackOverflow](https://stackoverflow.com/questions/tagged/flutter) or on one of the flutter community sites at https://flutter.dev/community

**Comment by ahyangnb** [comment]
Some people say it is preloading, but I still feel very unreasonable.

I preloaded the two adjacent ones, and then I switched one of the two adjacent ones, and then printed the adjacent tabbarView page that had already been printed.

I switched from 0 to 4 and then switched to 3 to print:
`` `
flutter: 0
flutter: 3
flutter: 4
flutter: 3
flutter: 3
`` `

All pages are configured with AutomaticKeepAliveClientMixin to save state is useless.
